### ISSUE TITLE
Lint `"pkg" %in% installed.packages()`

### ISSUE THREAD

**Issue by Bisaloo** [jimhester/lintr#2932]
``` r
lintr::lint(
  text = '"pkg" %in% installed.packages()', 
  linters = lintr::all_linters()
)
#> ℹ No lints found.
```

<sup>Created on 2025-09-16 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

`installed.packages()` can be pretty slow, as noted in the documentation:

> This needs to read several files per installed package, which will be slow on Windows and on some network-mounted file systems.
>
> It will be slow when thousands of packages are installed, so do not use it to find out if a named package is installed (use find.package or system.file) nor to find out if a package is usable (call requireNamespace or require and check the return value) nor to find details of a small number of packages (use packageDescription).

And the recommended way to test for package availability should be `requireNamespace("pkg", quietly = TRUE)`.

**Comment by MichaelChirico** [comment]
Sounds reasonable. I do worry about risk of false positives -- obviously there is _some_ good use for `installed.packages()`.

Would you mind sharing some good examples you find "in the wild" here?

BTW, in your title you use `"pkg" %in% installed.packages()`, which is definitely wrong vs. `"pkg" %in% rownams(installed.packages())`, is the latter what you meant? I do see a lot of people making the former mistake: https://github.com/search?q=lang%3AR+%2F%25in%25%5Cs*installed%5C.packages%5C%28%5C%29%5B%5E%5C%5B%5D%2F&type=code

**Comment by Bisaloo** [comment]
Yes, I noticed the incorrect `"pkg" %in% installed.packages()` in some examples. But by this I also meant linting `"pkg" %in% installed.packages()[, 1]` and `"pkg" %in% installed.packages()[, "Package"]` which are frequent alternatives to `"pkg" %in% rownames(installed.packages())`:

https://github.com/search?q=lang%3AR+%2F%25in%25%5Cs*installed%5C.packages%5C%28%5C%29%5C%5B%5Cs*%2C%5Cs*%281%7C%5B%22%27%5DPackage%5B%22%27%5D%29%5C%5D%2F&type=code

---

The function that prompted this issue:

https://github.com/Bioconductor/ExperimentHub/blob/f2a82b504abea3cf7eb9806afc5c12229a5c3e70/R/ExperimentHub-class.R#L90-L114
